crates/nu-utils/standard_library/tests.nu

```diff
@@ -37,20 +29,55 @@ def main [
     --command: string, # Test command to run. Default: all test command found in the files.
     --list, # Do not run any tests, just list them (dry run)
 ] {
-    let dry_run = ($list | default false)
-    for test_file in (collect-modules $path $module) {
-        let $module_name = ($test_file | path parse).stem
+    let tests = (
+        ls ($path | default $env.FILE_PWD | path join "test_*.nu")
+        | each {|row| {file: $row.name name: ($row.name | path parse | get stem)}}
+        | upsert test {|module|
+            nu -c $'use ($module.file) *; $nu.scope.commands | select name module_name | to nuon'
+            | from nuon
+            | where module_name == $module.name
+            | where ($it.name | str starts-with "test_")
+            | get name
+        }
+        | flatten
+        | rename file module name
+    )
 
-        log info $"Run tests in ($module_name)"
-        let tests = (collect-commands $test_file $module_name $command)
+    if $list {
+        return ($tests | select module name file)
+    }
 
-        for test_case in $tests {
-            log debug $"Run test ($module_name) ($test_case)"
-            if $dry_run {
-                continue
-            }
+    let tests_to_run = (if not ($command | is-empty) {
+        $tests | where name == $command
+    } else if not ($module | is-empty) {
+        $tests | where module == $module
+    } else {
+        $tests
+    })
 
-            nu -c $'use ($test_file) ($test_case); ($test_case)'
+    let tests = (
+        $tests_to_run | upsert pass {|test|
+            log info $"Run test ($test.module) ($test.name)"
```

Comment:
Do you want to log it in info level? Previously it was debug.

Reply:
I see, now it is harder to log the processed module in info level. I can accept it.

Reply:
should be good in 5c0ac96da3885165382408b9df493573084f04ff